### DISCUSSION TITLE
upgrade from 2.1.27.36  to  version 3.1.3 Docker

### DISCUSSION THREAD

**Discussion by noureddin212** [nilsteampassnet/TeamPass#4529] [Q&A]
Hello Guys 
Great Job about TeamPass .IT is a very powerful solution  Thanks already for that 
I have a question :
We have a 2.1.27.36 version we would like to upgrade to the last one by security we do not want to touch the production one .
So it is more a migration ( create a new TeamPAss with last version and import data from the production server )

I was thinking to 1 install a docker Teampass vith last version 
and import the forlder/DB to that docker .

Is it the good way to doit if so is there a doc that I can follow or should I do it another way 

Thanks for your help

**Comment by mkocbek** [comment]
Stay with current version. New versions are not production ready in my opinion. Been using Teampass for 10 years at multiple clients. Really great software but so many bugs at the moment.  

Upgraded to new version. Just to upgrade i needed to change source code multiple times. Manually fix database multiple times.

**Reply by corentin-soriano** [reply]
The update process had a lot of fixes that were merged in master branch early September.
This resolved issues with database fields, errors during updates, and big improvements in installation times.

I needed a little less than 4 hours for a database with 600 users and 30,000 items (against an estimate > 90 hours before).

Don't hesitate to try the upgrade with a copy of your teampass 2 database on another server.
